frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -299,8 +299,23 @@ export const experimentLogic = kea<experimentLogicType>([
     }),
     listeners(({ values, actions }) => ({
         createExperiment: async ({ draft, runningTime, sampleSize }) => {
+            const { experiment, experimentInsightType } = values
             let response: Experiment | null = null
             const isUpdate = !!values.experimentId && values.experimentId !== 'new'
+
+            if (experimentInsightType === InsightType.FUNNELS) {
```

Comment:
looking deeper, this is orthogonal to the https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/experiments/experimentLogic.tsx#L466 updateExperimentGoal function which has this same problem.

Would be nicer imo to make this part of the form validation, which should automatically apply to both places, because metricSelector is embedded in the form -> https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/experiments/ExperimentView/Goal.tsx#L125

So it's a form error, instead of a toast error on submit. And since we have filters in the form, should be reasonably easy to handle there as well. And hopefully, the errors would by default show up in the right place, given how things are setup